lib/filters/shoutcloud.coffee

```diff
@@ -0,0 +1,29 @@
+request = require 'request'
+
+module.exports =
+  name: "ShoutCloud"
+  description: ""
+
+  applies: (req) ->
+    req.query? and req.query.shoutcloud?
+
+  process: (req, res, message, subtitle, next) =>
+
+    str = "#{message}**seperator**#{subtitle}"
+
+    request.post {
+      headers: 'content-type': 'application/json'
+      url: 'HTTP://API.SHOUTCLOUD.IO/V1/SHOUT'
+      body: '{"INPUT": "'+str+'"}'
+    }, (error, response, body) =>
+      return @onError(req, res) if error?
+      try
+        str = JSON.parse(body).body['OUTPUT'].split('**SEPERATOR**')
+        next(req,res,str[0],str[1])
+      catch
+        return @onError(req, res)
+
+  onError: (req,res) ->
+    res.status 408
+    res.send('408 Bad Gateway - Upstream soundcloud.io is down.')
```

Comment:
Well, if the JSON is bad then the upstream is bad
On 7 Feb 2015 20:46, "Mark Rickerby" <notifications@github.com> wrote:

> In lib/filters/shoutcloud.coffee
> <https://github.com/tomdionysus/foaas/pull/69#discussion_r24288460>:
>
> > +
> > +    request.post {
> > +      headers: 'content-type': 'application/json'
> > +      url: 'HTTP://API.SHOUTCLOUD.IO/V1/SHOUT'
> > +      body: '{"INPUT": "'+str+'"}'
> > +    }, (error, response, body) =>
> > +      return @onError(req, res) if error?
> > +      try
> > +        str = JSON.parse(body).body['OUTPUT'].split('**SEPERATOR**')
> > +        next(req,res,str[0],str[1])
> > +      catch
> > +        return @onError(req, res)
> > +
> > +  onError: (req,res) ->
> > +    res.status 408
> > +    res.send('408 Bad Gateway - Upstream soundcloud.io is down.')
>
> Also you’re blaming JSON parsing errors on the upstream HTTP service here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tomdionysus/foaas/pull/69/files#r24288460>.
>